src/colorizer/ColorizeCanvas.ts

```diff
@@ -529,22 +537,63 @@ export default class ColorizeCanvas {
     return index >= 0 && index < this.getTotalFrames();
   }
 
+  public setObjectOpacity(percentOpacity: number): void {
+    percentOpacity = Math.max(0, Math.min(100, percentOpacity));
+    this.setUniform("objectOpacity", percentOpacity / 100);
+  }
+
+  public setBackdropSaturation(percentSaturation: number): void {
+    percentSaturation = Math.max(0, Math.min(100, percentSaturation));
+    this.setUniform("backdropSaturation", percentSaturation / 100);
+  }
+
+  public setBackdropBrightness(percentBrightness: number): void {
+    percentBrightness = Math.max(0, Math.min(200, percentBrightness));
+    this.setUniform("backdropBrightness", percentBrightness / 100);
+  }
+
+  public setBackdropKey(backdropKey: string | null): void {
+    this.backdropKey = backdropKey;
+    this.setFrame(this.currentFrame, true).then(() => {
+      this.render();
+    });
+  }
+
   /**
    * Sets the current frame of the canvas, loading the new frame data if the
    * frame number changes.
    * @param index Index of the new frame.
+   * @param forceUpdate Force a reload of the frame data, even if the frame
+   * is already loaded.
    */
-  async setFrame(index: number): Promise<void> {
+  async setFrame(index: number, forceUpdate: boolean = false): Promise<void> {
     // Ignore same or bad frame indices
-    if (this.currentFrame === index || !this.isValidFrame(index)) {
+    if ((!forceUpdate && this.currentFrame === index) || !this.isValidFrame(index)) {
       return;
     }
     // New frame, so load the frame data.
     this.currentFrame = index;
-    const frame = await this.dataset?.loadFrame(index);
+    let backdropPromise = undefined;
+    if (this.backdropKey && this.dataset?.hasBackdrop(this.backdropKey)) {
+      backdropPromise = await this.dataset?.loadBackdrop(this.backdropKey, index);
+    }
+    const framePromise = await this.dataset?.loadFrame(index);
+    const result = await Promise.all([framePromise, backdropPromise]);
```

Comment:
It looks like this result might be unnecessary because you already awaited both calls?

Reply:
Oops! Good catch, thanks. Removed the two `awaits` for each promise so they can run in parallel.